src/server/Adaptive.ReactiveTrader.Server.Bot/src/index.ts

```diff
@@ -53,23 +60,80 @@ const symphony = new SymphonyClient(symphonyConfig, key, false)
 
 const priceToQuery = (prices: Map<string, Price>, text) => Array.from(prices.keys()).find(pair => text.search(pair) > -1)
 
-const messageEvent$ = symphony.dataEvents$()
+const logSubscriber: NextObserver<any> = {
+  next: message => logger.info('Messaged succesfully sent to Syphony'),
+  error: (error) => logger.error('Error sending to Syphony', error)
+}
+
+const messageEvent$ = symphony.dataEvents$().pipe(share())
+
+const priceSubscriber$ = messageEvent$.pipe(
+  filter(message => message.messageText.includes('price')),
+  withLatestFrom(priceSubsription$),
+  mergeMap(([message, latestPrices]) => {
+    const matched = priceToQuery(latestPrices, message.messageText)
+    const messageMarkup = matched ? createPriceMessage(latestPrices.get(matched)!) : `No symbol found for: ${message.messageText}`
+    return symphony.sendMessage(message.stream.streamId, messageMarkup)
+  })
+).subscribe(logSubscriber)
+
+const marketSubscriber$ = messageEvent$.pipe(
+  filter(message => message.messageText.includes('market')),
+  withLatestFrom(priceSubsription$),
+  mergeMap(([message, latestPrices]) => {
+    const messageMarkup = marketUpdateMessage(Array.from(latestPrices.values()))
+    return symphony.sendMessage(message.stream.streamId, messageMarkup)
+  })
+).subscribe(logSubscriber)
+
+const statusUpdates$ = stub.subscribeToTopic<RawServiceStatus>('status')
+const serviceStatus$ = serviceStatusStream$(statusUpdates$, HEARTBEAT_TIMEOUT).pipe(
+  multicast(() => {
+    return new ReplaySubject<ServiceCollectionMap>(1)
+  }),
+  refCount(),
+)
+
+const loadBalancedServiceStub = new ServiceStubWithLoadBalancer(stub, serviceStatus$)
+
+const tradeStream$ = loadBalancedServiceStub.createStreamOperation<TradeUpdate>('blotter', 'getTradesStream', {}).pipe(share());
+
+const latestTrades$ = tradeStream$
   .pipe(
-    tap(message => logger.info(message)),
-    withLatestFrom(priceSubsription$),
-    mergeMap(([message, latestPrices]) => {
-      const matched = priceToQuery(latestPrices, message.messageText)
-      const returnMessage = matched ? createPriceMessage(latestPrices.get(matched)!) : `No symbol found for: ${message.messageText}`
-      return symphony.sendMessage(message.stream.streamId, returnMessage)
-    })
-  ).subscribe(message => {
-    logger.info('Messaged sent to Syphony', message)
-  },
-    (error) => {
-      logger.error('Error sending to Syphony', error)
-    }
+    map(tradeUpdate => tradeUpdate.Trades),
+
+    scan<Trade[], Map<number, Trade>>((acc, trades) => {
+      trades.forEach(trade => acc.set(trade.TradeId, trade))
+      return acc
+    },
+      new Map<number, Trade>()),
+    map(trades => Array.from(trades.values())),
+    share()
   )
 
+
+const trade$ = messageEvent$.pipe(
+  filter(message => message.messageText.includes('trades')),
```

Comment:
Not needed right away, but I do think a way to unsubscribe from trade updates once requested is important.